src/internal_modules/roc_audio/depacketizer.cpp

```diff
@@ -107,21 +108,36 @@ Depacketizer::read_samples_(sample_t* buff_ptr, sample_t* buff_end, FrameInfo& i
                 * (size_t)packet::timestamp_diff(next_timestamp, timestamp_);
 
             const size_t max_samples = (size_t)(buff_end - buff_ptr);
+            const size_t nsamples = std::min(mis_samples, max_samples);
 
-            buff_ptr = read_missing_samples_(
-                buff_ptr, buff_ptr + std::min(mis_samples, max_samples));
+            buff_ptr = read_missing_samples_(buff_ptr, buff_ptr + nsamples);
+            if (!info.capture_ts && valid_capture_ts_) {
+                info.capture_ts =
+                    next_capture_ts_ - sample_spec_.samples_overall_2_ns(mis_samples);
+            }
         }
 
         if (buff_ptr < buff_end) {
             sample_t* new_buff_ptr = read_packet_samples_(buff_ptr, buff_end);
-
-            info.n_decoded_samples += size_t(new_buff_ptr - buff_ptr);
-
+            const size_t nsamples = size_t(new_buff_ptr - buff_ptr);
+
+            info.n_decoded_samples += nsamples;
+            if (nsamples && !info.capture_ts && valid_capture_ts_) {
+                info.capture_ts = next_capture_ts_;
+            }
+            if (valid_capture_ts_) {
+                next_capture_ts_ += sample_spec_.samples_overall_2_ns(nsamples);
+            }
             buff_ptr = new_buff_ptr;
         }
 
         return buff_ptr;
     } else {
+        const size_t nsamples = size_t(buff_end - buff_ptr);
+        if (valid_capture_ts_) {
+            info.capture_ts = next_capture_ts_;
+            next_capture_ts_ += sample_spec_.samples_overall_2_ns(nsamples);
+        }
```

Comment:
Don't we need:

```
if (!info.capture_ts) {
    info.capture_ts = next_capture_ts_;
}
```

?

Reply:
No, assertion is enough:

```
const size_t nsamples = size_t(buff_end - buff_ptr);
if (valid_capture_ts_) {
    roc_panic_if(info.capture_ts);
    info.capture_ts = next_capture_ts_;
    next_capture_ts_ += sample_spec_.samples_overall_2_ns(nsamples);
}
return read_missing_samples_(buff_ptr, buff_end);
```

As we fill all the requested samples, the outer while loop will never run the second iteration.

Reply:
My bad, assertion fails. We have to check this!

Reply:
@gavv  Augmented `TEST(depacketizer, frame_flags_incompltete_blank)` to cover this edge-case, and stumbled upon probable bug there, pls take a look